src/zarr.hh

```diff
@@ -132,6 +134,27 @@ struct Zarr final : StorageInterface
     void recover_threads_();
 };
 
+struct ZarrV3 final : public Zarr
```

Comment:
Optional: don't feel strongly enough to really suggest this, but I wonder if we should really use inheritance here. Seems like it's mostly being used to share code rather storing heterogenous collections of zarr and zarrv3 objects and I generally feel like plain old functions are typically better for sharing code.